File: bench/clojure/spawn.clj
;; fan out N concurrent units, each computing fib(15); sum the results.
;; Clojure's idiomatic lightweight concurrency = futures (JVM thread pool).
(def n (Long/parseLong (or (System/getenv "BENCH_N") "5000")))
(defn fib [x] (if (< x 2) x (+ (fib (- x 1)) (fib (- x 2)))))
(println (reduce + (map deref (doall (map (fn [_] (future (fib 15))) (range n))))))

;; Stop the agent/future thread pool so the JVM exits promptly (else it lingers ~60s).
(shutdown-agents)

File: bench/go/spawn.go
// Fan out N goroutines; each computes fib(15) and sends the result back on a channel.
// Checksum = N * fib(15) = N * 610.
package main

import "fmt"

func fib(n int) int {
	if n < 2 {
		return n
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	n := benchN(10000)
	results := make(chan int, n)
	for i := 0; i < n; i++ {
		go func() { results <- fib(15) }()
	}
	total := 0
	for i := 0; i < n; i++ {
		total += <-results
	}
	fmt.Println(total)
}
